code/modules/clothing/suits/reactive_armour.dm

```diff
@@ -40,102 +52,306 @@
+		owner.visible_message("<span class='danger'>[src] attempts to react, but the flame jets are blocked by what [owner] is holding!</span>")
+		return FALSE
+	owner.visible_message("<span class='danger'>[src] blocks [attack_text], opening the flame jets!</span>")
+	return TRUE
+
+/obj/item/clothing/suit/armor/reactive/fire/emp_activation(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	owner.visible_message("<span class='danger'>[src] just makes [attack_text] worse by jamming and rupturing, releasing molten death on [owner]!</span>")
+	playsound(get_turf(owner),'sound/magic/fireball.ogg', 100, TRUE)
+	owner.adjust_fire_stacks(12)
+	owner.IgniteMob()
+	reactivearmor_cooldown = world.time + reactivearmor_cooldown_duration
 	return FALSE
 
+/obj/item/clothing/gloves/flame_jets
+	name = "reactive incendiary gauntlets"
+	desc = "A pair of metal gauntlets that are hooked up to the reactive incendiary armor. When threatened, you will be able to spew flame from them... In theory!"
+	icon_state = "flamejet"
+	permeability_coefficient = 0.9
+	heat_protection = HANDS
+	max_heat_protection_temperature = GLOVES_MAX_TEMP_PROTECT
+	resistance_flags = NONE
+	clothing_traits = list(TRAIT_CHUNKYFINGERS)// gaunts are much too big to be able to mess with stuff like that
+	armor = list(MELEE = 0, BULLET = 0, LASER = 0, ENERGY = 0, BOMB = 0, BIO = 0, RAD = 0, FIRE = 100, ACID = 30)
+	var/obj/item/clothing/suit/armor/reactive/fire/connected_armor
+
+/obj/item/clothing/gloves/flame_jets/Initialize()
+	. = ..()
+	ADD_TRAIT(src, TRAIT_NODROP, "flame jets")
+
+/obj/item/clothing/gloves/flame_jets/dropped(mob/user)
+	. = ..()
+	if(ismob(loc))
+		return
+	//something like dismemberment happened
+	if(!connected_armor)
+		qdel(src)
+		return
+	connected_armor.attack_self(user) //disengage the gloves.
+
+/obj/item/gun/ballistic/rifle/boltaction/enchanted/arcane_barrage/flame_jets
+	name = "active flame jets"
+	desc = "Taking \"reactive armor\" into your own hands."
+	fire_sound = 'sound/magic/fireball.ogg'
+	icon_state = "flamejet"
+	pin = /obj/item/firing_pin/fuelline
+	guns_left = 5 //6 shots in total
+	inhand_icon_state = "flamejet"
+
+	mag_type = /obj/item/ammo_box/magazine/internal/boltaction/enchanted/flame_jet
+
+/obj/item/gun/ballistic/rifle/boltaction/enchanted/arcane_barrage/flame_jets/Initialize()
+	. = ..()
+	QDEL_IN(src, 6 SECONDS)
+
+/obj/item/ammo_box/magazine/internal/boltaction/enchanted/flame_jet
+	ammo_type = /obj/item/ammo_casing/magic/flame_jet
+
+/obj/item/ammo_casing/magic/flame_jet
+	projectile_type = /obj/projectile/flame_jet
+
+/obj/projectile/flame_jet
+	name = "flame jet"
+	icon_state = "pulse0"
+	damage = 5
+	damage_type = BURN
+	armour_penetration = 0
+	hitsound = 'sound/weapons/barragespellhit.ogg'
+
+/obj/projectile/flame_jet/on_hit(target)
+	if(isliving(target))
+		var/mob/living/burning_living = target
+		burning_living.adjust_fire_stacks(2)
```

Comment:
holy shit this is bad

why would I be firing 5 damage 2 fire stack projectiles that come from a weapon that occupies my exosuit slot AND my gloves slot instead of using literally anything else

Reply:
I would seriously recommend just going for something much simpler and less over-engineered, like the "attacker receives the fire hallucination" idea mentioned earlier.